modules/qtx_modules_handler.php

```diff
@@ -73,6 +75,7 @@ public static function update_manual_enabled_modules() {
         if ( $changed ) {
             update_option( 'qtranslate_modules', $options_modules );
             self::load_active_modules();
+            // TODO remove this action
```

Comment:
We need to give opportunity to modules to perform activation/deactivation actions (e.g. flush rewrite rules for slugs module) on manual state change.
I don't see other clean ways than a `do_action` here...

Reply:
I know it's needed, but using a hook for something purely internal is a *hack*. No need to fix this now but it can *always* be designed differently ;)
The hooks are the main reason why WordPress became so popular, and one of the reasons this framework is such a huge mess.